jans-cedarling/cedarling/src/init/policy_store.rs

```diff
@@ -0,0 +1,58 @@
+/*
+ * This software is available under the Apache-2.0 license.
+ * See https://www.apache.org/licenses/LICENSE-2.0.txt for full text.
+ *
+ * Copyright (c) 2024, Gluu, Inc.
+ */
+
+use crate::models::policy_store::{PolicyStore, PolicyStoreMap};
+use crate::models::policy_store_config::{PolicyStoreConfig, PolicyStoreSource};
+
+/// Error cases for loading policy
+#[derive(Debug, thiserror::Error)]
+pub enum ErrorLoadPolicyStore {
```

Comment:
Rather than creating different enums for showing error messages, can we just make a single enum for all error messages inside `\jans-cedarling\cedarling\src\models\enum\error`.

**for Example:**
```
pub enum CedarlingApplicationError {
 #[error("{0}")]
    LoadPolicyStoreJsonParseError(#[from] serde_json::Error),
    #[error("store policy is empty")]
    LoadPolicyStoreWithEmptyPolicy,
    #[error("the `store_key` is not specified and the count on policies more than 1")]
    LoadPolicyStoreMoreThanOnePolicy,
    #[error("could not found policy by id: {0}")]
    LoadPolicyStorePolicyNotFound(String),
}
```

Reply:
I like the idea to return the error struct in functions that cover all possible error cases for only the current situation. Because it will be easier to maintain when application grows.

And If we have somethinglike `CedarlingApplicationError`
It will have all possible errors for all application. 

And when we will be wanted to handle the error cases with match statement, it will be hard to understand what exactly cases current function returns. Because `CedarlingApplicationError` have all possible cases in the application.

Reply:
> And If we have something like CedarlingApplicationError
It will have all possible errors for all application.

This is what we practise in most of applications. We do not clutter our project with multiple Enums for error messages.

> And when we will be wanted to handle the error cases with match statement, it will be hard to understand what exactly cases current function returns. Because CedarlingApplicationError have all possible cases in the application.

The naming convention of enum items should be sufficient of make it clear purpose. We don't need to make multiple error enums for this.